### ISSUE TITLE
[java] Duplicate warning when enabling deprecated rule

### ISSUE THREAD

**Issue by davewichers** [pmd/pmd#6610]
Not sure if this is specific to Java or not. But I'm analyzing Java.

**Affects PMD Version:**

maven-pmd-plugin:3.28.0 with net.sourceforge.pmd:pmd-core and pmd-java overridden to be version: 7.23.0, instead of default 7.17.0

**Description:**

I'm reenabling the deprecated AvoidCatchingNPE rule, and getting this warning TWICE when running PMD.
```
[WARNING] Warning at E:\git\testcases\Java\target\pmd\rulesets\001-pmd-ruleset.xml:15:5
 13|         TODO: For PMD 8.0.0+, need to configure AvoidCatchingGenericException to flag
 14|               use of catching NullPointerException. -->
 15|     <rule ref="category/java/errorprone.xml/AvoidCatchingNPE"/>
         ^^^^^ Discontinue using Rule name category/java/errorprone.xml/AvoidCatchingNPE as it is scheduled for removal from PMD. PMD 8.0.0 will remove support for this Rule.

 16|     <rule ref="category/java/multithreading.xml"></rule>
 17|     <rule ref="category/java/performance.xml"></rule>
[WARNING] Warning at E:\git\testcases\Java\target\pmd\rulesets\001-pmd-ruleset.xml:15:5
 13|         TODO: For PMD 8.0.0+, need to configure AvoidCatchingGenericException to flag
 14|               use of catching NullPointerException. -->
 15|     <rule ref="category/java/errorprone.xml/AvoidCatchingNPE"/>
         ^^^^^ Discontinue using Rule name category/java/errorprone.xml/AvoidCatchingNPE as it is scheduled for removal from PMD. PMD 8.0.0 will remove support for this Rule.

 16|     <rule ref="category/java/multithreading.xml"></rule>
 17|     <rule ref="category/java/performance.xml"></rule>
```

**Steps to reproduce:**

Just enable this deprecated rule by adding:
`<rule ref="category/java/errorprone.xml/AvoidCatchingNPE"/>`
To your custom ruleset to use.

**Running PMD through:**

Maven

**Comment by UncleOwen** [comment]
Looking at the [source of maven-pmd-plugin](https://github.com/apache/maven-pmd-plugin/blob/5ba497aa1c133ba8647a64439b25a4b50b5f3c62/src/main/java/org/apache/maven/plugins/pmd/exec/PmdExecutor.java#L281) I see
```java
        try {
            // load the ruleset once to log out any deprecated rules as warnings
            rulesetLoader.loadFromResources(pmdConfiguration.getRuleSetPaths());
        } catch (RuleSetLoadException e1) {
            throw new MavenReportException("The ruleset could not be loaded", e1);
        }

        try (PmdAnalysis pmdAnalysis = PmdAnalysis.create(pmdConfiguration)) {
```
`PmdAnalysis.create()` ends up loading the ruleset again - and triggering the warning again.

This code has been there since [MPMD-268](https://github.com/apache/maven-pmd-plugin/issues/270). @adangel Do you remember why this was added? Didn't PMD log deprecated rules back then or something?

**Comment by UncleOwen** [comment]
Done via https://github.com/apache/maven-pmd-plugin/pull/703
